Integration accuracy test (inspired by L2Projection integration method)

aprokop opened the pull request:
Continuation of #119 after unsuccessful force push.

dalg24-jenkins commented:
Can one of the admins verify this patch?

aprokop commented:
ok to test

Rombur commented:
Jenkins didn't start :-/  Do a dummy push to the branch to trigger Jenkins again.

aprokop commented:
I wonder if it only checks for @dalg24's "ok to test". Could you try it too?

Rombur commented:
I can try but normally the test is triggered when the PR is created and by any push on a PR.

Rombur commented:
ok to test

dalg24 commented:
ok to test

aprokop commented:
Hmm, so apparently only @dalg24 can kick off the testing (at least for my PR)...

Rombur commented:
well that sucks...

aprokop commented:
I think this is a Jenkins configuration issue. Specifically, when #129 was submitted the Jenkins did not ask "can I test?" question. So it must be checking of who is submitting the patch.

dalg24 commented:
@aprokop What do you mean unsuccessful force push?  You are woking your own fork, you should be able to do that as often as you like...

aprokop commented:
@dalg24 I updated the master and rebased the fork on top of it, that broke it. I think as long as the commit you use as your base is the same it's fine; as soon as it changes github closes it. 
I tried rebasing on top of the old hash but that did not work for me either.

dalg24 commented:
That's weird I do this all the time and I never had any issue.

aprokop commented:
@sslattery @dalg24 Please review.

sslattery commented on packages/Operators/test/reference_implementation/DTK_ReferenceHexMesh.cpp:
```diff
@@ -208,53 +248,19 @@ void ReferenceHexMesh::buildMesh(
-                          ( k + 1 ) * x_num_nodes * y_num_nodes;
-                DTK_CHECK( node_id < total_nodes );
-                entity_set->getEntity( node_id, 0, hex_nodes[7] );
+#define GET_NODE( local_id, i, j, k )                                          \
```
Why not create a function instead of using a macro?

sslattery replied:
or a lambda?

aprokop replied:
Don't think a function is a good choice. But did not think about lambdas. Will try to implement those.

sslattery commented on packages/Operators/test/tstIntegral.cpp:
```diff
@@ -0,0 +1,472 @@
+    }
+
+    // should not reach this
+    return 0.0;
```
maybe throw then if we reach this point?

aprokop replied:
OK.

aprokop commented on packages/Operators/test/reference_implementation/DTK_ReferenceHexMesh.cpp:
```diff
@@ -50,8 +52,6 @@
 #include <DTK_BasicEntitySet.hpp>
 #include <DTK_EntityCenteredField.hpp>
 
-#include <DTK_DBC.hpp>
```
This was moved to `.hpp` file. It probably should have been a part of the commit containing `id()` calls as to make them inline I put them into `hpp` while simultaneously doing some `DTK_REQUIRE` there.

aprokop commented:
Updated PR to address @sslattery comments.

dalg24 commented:
That's weird I had these "pending comments".

aprokop commented on packages/Operators/test/tstL2Projection.cpp:
```diff
@@ -96,17 +96,20 @@ double integrateField( DataTransferKit::UnitTest::ReferenceHexMesh &mesh,
                    DataTransferKit::FieldMultiVector::GO>
         importer( vector->getMap(), ghosted_vector->getMap() );
     ghosted_vector->doImport( *vector, importer, Tpetra::REPLACE );
+    ghosted_vector->pushDataToApplication();
```
You need this to sync data in MultiVector and field.
